src/audio/player.js

```diff
@@ -316,1 +1,316 @@
+            rawData,
+        }];
+
+        if (this._songs.length === 0) {
+            this._songs.push(...parsedData);
+            this.playStream();
+        } else {
+            this._songs.push(...parsedData);
+        }
+
+        this._player.once(AudioPlayerStatus.Playing, () => {
+            log.info(`${this._guildId}:${this._channelId} 音樂播放器進入播放狀態`);
+            this.handelPlaying();
+        });
+    }
+
+    skip() {
+        this._songs.shift();
+        const skipEmbed = new Discord.MessageEmbed()
+            .setTitle(`⏭️ 跳過歌曲 **${this._audio.metadata.title}**`)
+            .setColor(colors.success);
+        this._channel.send({
+            embeds: [skipEmbed],
+        }).catch(this.noop);
+        this._player.stop();
+    }
+
+    pause() {
+        this._player.pause();
+        const pauseEmbed = new Discord.MessageEmbed()
+            .setTitle('⏸️ 暫停音樂')
+            .setColor(colors.success);
+        this._channel.send({
+            embeds: [pauseEmbed],
+        }).catch(this.noop);
+    }
+
+    unpause() {
+        this._player.unpause();
+        const unpauseEmbed = new Discord.MessageEmbed()
+            .setTitle('▶️ 繼續播放音樂')
+            .setColor(colors.success);
+        this._channel.send({
+            embeds: [unpauseEmbed],
+        }).catch(this.noop);
+    }
+
+    shuffle() {
+        const shuffled = [].concat(this._songs);
+        let currentIndex = array.length, temporaryValue, randomIndex;
+
+        while (currentIndex !== 0) {
+            randomIndex = Math.floor(Math.random() * currentIndex);
+            currentIndex -= 1;
+
+            temporaryValue = shuffled[currentIndex];
+            shuffled[currentIndex] = shuffled[randomIndex];
+            shuffled[randomIndex] = temporaryValue;
+        }
+
+        const unpauseEmbed = new Discord.MessageEmbed()
+            .setTitle('🔀 重新排序音樂')
+            .setColor(colors.success);
+        this._channel.send({
+            embeds: [unpauseEmbed],
+        }).catch(this.noop);
+        this.songs = shuffled;
+    }
+
+    async playStream() {
+        if (!this._songs[0]?.rawData.full) {
+            try {
+                this._songs[0].rawData = await play.video_info(this._songs[0].url);
+                this._songs[0].rawData.full = true;
+            } catch (e) {
+                log.error(e.message);
+                const errorEmbed = new Discord.MessageEmbed()
+                    .setTitle('🙁 載入音樂時發生錯誤')
+                    .setDescription(
+                        '載入音樂時發生了一點小錯誤...\n' +
+            '錯誤內容:\n' +
+            '```\n' + e.message + '\n```')
+                    .setColor(colors.danger);
+                this._channel.send({
+                    embeds: [errorEmbed],
+                }).catch(this.noop);
+                return;
+            }
+        }
+
+        let stream;
+        try {
+            stream = await play.stream(this._songs[0].url);

```

Comment:
Codacy has a fix for the issue: [Expected indentation of 12 spaces but found 6.](https://app.codacy.com/gh/blackcatbot/blackcat/pullRequest?prid=8779465)
```suggestion
            stream = await play.stream(this._songs[0].url);
```